src/Http/Middleware/RateLimit.php

```diff
@@ -64,7 +64,8 @@ public function handle($request, Closure $next)
         $this->handler->rateLimitRequest($request, $route->getRateLimit(), $route->getRateLimitExpiration());
 
         if ($this->handler->exceededRateLimit()) {
-            throw new HttpException(403, 'You have exceeded your rate limit.', null, $this->getHeaders());
+            $retryAfter = $this->getHeaders()['X-RateLimit-Reset'] - time();
+            throw new TooManyRequestsHttpException($retryAfter, 'You have exceeded your rate limit.');
```

Comment:
@hskrasek & @lucasmichot - The reason i wanted to specifically throw the correct named exception was so it could be caught in custom exception responses. It is not really feasible to catch all thrown HttpExceptions (at least not in my use case) and modify them, i guess..

Is it maybe possible to create a custom HTTP exception that implements HttpExceptionInterface so others are able to hook into it and define a custom response? What are everyones thoughts?